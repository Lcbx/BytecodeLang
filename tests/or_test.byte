4<6 or 45==45 or 23==32
'should jump to line 3 after 4<6'
23==32 or 45==46 or 4<6
'should execute all line 3'
23==23 or 45==45 or 4<6
'should jump to value 1024 after 45 == 45'
33 == 11 * 3 or 23 == 23
'should jump to value 1024 after 33 == 11 * 3'
1024