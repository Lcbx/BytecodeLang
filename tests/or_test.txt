hi 1012 +12
hi 4<6 or 45==45 or 23==32
hi 1024
hi 23==32 or 45==46 or 4<6
hi 1024
hi 23==32 or 45==45 or 4<6
hi 1024
hi 23==23 and 45==46 or 4<6
hi 1024