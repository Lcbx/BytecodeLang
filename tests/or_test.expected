tokenize_Out:
INT: 4, OP: <, INT: 6, NAME: or, INT: 45, OP: ==, INT: 45, NAME: or, INT: 23, OP: ==, INT: 32, STRING: should jump to line 3 after 4<6, INT: 23, OP: ==, INT: 32, NAME: or, INT: 45, OP: ==, INT: 46, NAME: or, INT: 4, OP: <, INT: 6, STRING: should execute all line 3, INT: 23, OP: ==, INT: 23, NAME: or, INT: 45, OP: ==, INT: 45, NAME: or, INT: 4, OP: <, INT: 6, STRING: should jump to value 1024 after 45 == 45, INT: 33, OP: ==, INT: 11, OP: *, INT: 3, NAME: or, INT: 23, OP: ==, INT: 23, STRING: should jump to value 1024 after 33 == 11 * 3, INT: 1024, EOF: line 8
__SECTION__
tokenize_Err:

__SECTION__
compile_Out:
0 errors
__SECTION__
compile_Err:

__SECTION__
simulation_Out:
0    	OP_INT1 => 4 --
2    	OP_INT1 => 6 --
4    	OP_LT => True --
5    	OP_JUMP_IF => jump 15 --
23   	OP_STRING => should jump to line 3 after 4<6 --
56   	OP_INT1 => 23 --
58   	OP_INT1 => 32 --
60   	OP_EQ => False --
61   	OP_JUMP_IF => !jump 15 --
64   	OP_POP => False --
65   	OP_INT1 => 45 --
67   	OP_INT1 => 46 --
69   	OP_EQ => False --
70   	OP_JUMP_IF => !jump 6 --
73   	OP_POP => False --
74   	OP_INT1 => 4 --
76   	OP_INT1 => 6 --
78   	OP_LT => True --
79   	OP_STRING => should execute all line 3 --
106  	OP_INT1 => 23 --
108  	OP_INT1 => 23 --
110  	OP_EQ => True --
111  	OP_JUMP_IF => jump 15 --
129  	OP_STRING => should jump to value 1024 after 45 == 45 --
171  	OP_INT1 => 33 --
173  	OP_INT1 => 11 --
175  	OP_INT1 => 3 --
177  	OP_MUL => 33 --
178  	OP_EQ => True --
179  	OP_JUMP_IF => jump 6 --
188  	OP_STRING => should jump to value 1024 after 33 == 11 * 3 --
234  	OP_INT2 => 1024 --
__SECTION__
simulation_Err:
